From c164029a24650c8bcfc9658e86a900fe72735682 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Wed, 22 Jan 2014 10:47:44 +0800
Subject: [PATCH 286/304] ENGR00296033 [fsltool] : fsl-sdcard-partition.sh
 can't work

Add patition 'p' after device name

Squashed commit of the following:

commit 91d5103bf0d85949f9fba034c8f81acc43309758
Author: Erik Larsson <erik@ortogonal.com>
Date:   Mon Jan 20 17:28:11 2014 +0100

    Bug fix: fsl-sdcard-partition.sh didn't handle /dev/mmcblk devices correct

    The script had a variable that should a part as p if a /dev/mmcblk device
    was used. But it only checked the diskname and did not used the variable
    later on.

    Signed-off-by: Erik Larsson <erik@ortogonal.com>

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 common/tools/fsl-sdcard-partition.sh | 23 +++++++++++------------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/common/tools/fsl-sdcard-partition.sh b/common/tools/fsl-sdcard-partition.sh
index 2012acb..8c33b20 100755
--- a/common/tools/fsl-sdcard-partition.sh
+++ b/common/tools/fsl-sdcard-partition.sh
@@ -83,13 +83,13 @@ fi
 function format_android
 {
     echo "formating android images"
-    mkfs.ext4 ${node}4 -Ldata
-    mkfs.ext4 ${node}5 -Lsystem
-    mkfs.ext4 ${node}6 -Lcache
-    mkfs.ext4 ${node}7 -Lvender
+    mkfs.ext4 ${node}${part}4 -Ldata
+    mkfs.ext4 ${node}${part}5 -Lsystem
+    mkfs.ext4 ${node}${part}6 -Lcache
+    mkfs.ext4 ${node}${part}7 -Lvender
     mkdir /media/tmp
-    mount ${node}4 /media/tmp
-    amount=$(df -k | grep ${node}4 | awk '{print $2}')
+    mount ${node}${part}4 /media/tmp
+    amount=$(df -k | grep ${node}${part}4 | awk '{print $2}')
     stag=$amount
     stag=$((stag-32))
     kilo=K
@@ -97,8 +97,8 @@ function format_android
     sleep 1s
     umount /media/tmp
     rm -rf /media/tmp
-    e2fsck -f ${node}4
-    resize2fs ${node}4 $amountkilo
+    e2fsck -f ${node}${part}4
+    resize2fs ${node}${part}4 $amountkilo
 }
 
 function flash_android
@@ -107,9 +107,9 @@ if [ "${flash_images}" -eq "1" ]; then
     echo "flashing android images..."    
     dd if=u-boot.bin of=${node} bs=1k seek=1 skip=1
     dd if=/dev/zero of=${node} bs=512 seek=1536 count=16
-    dd if=boot.img of=${node}1
-    dd if=recovery.img of=${node}2
-    dd if=system.img of=${node}5
+    dd if=boot.img of=${node}${part}1
+    dd if=recovery.img of=${node}${part}2
+    dd if=system.img of=${node}${part}5
 fi
 }
 
@@ -118,7 +118,6 @@ if [[ "${not_partition}" -eq "1" && "${flash_images}" -eq "1" ]] ; then
     exit
 fi
 
-
 # destroy the partition table
 dd if=/dev/zero of=${node} bs=1024 count=1
 
-- 
1.8.0

